### PR TITLE
Normalize using Compat.

### PR DESCRIPTION
**Pull request by amitjamadagni**

### CODE REVIEW

**Comment by amitjamadagni** [comment]
@acroy `normalize` is breaking other packages (for example https://github.com/JuliaQuantum/QuDynamics.jl/pull/42), it would be helpful if we could have a quick review on this. Thanks !

File: src/arrays/arraymath.jl [hunk 1]
```diff
@@ -143,12 +143,12 @@ Base.dot{B<:OrthonormalBasis}(qm1::AbstractQuMatrix{B}, qm2::DualMatrix{B}) = tr
 # normalization
 Base.norm(qarr::AbstractQuArray) = vecnorm(rawcoeffs(qarr))
 
-function normalize!(qarr::AbstractQuArray)
+function Compat.normalize!(qarr::AbstractQuArray)
```
Comment (acroy) [inline src/arrays/arraymath.jl]: Does Compat define `normalize!` as well?

Comment (amitjamadagni) [inline src/arrays/arraymath.jl]: Yup it is mentioned in the README of the package https://github.com/JuliaLang/Compat.jl

Comment (acroy) [inline src/arrays/arraymath.jl]: Ok. Then this should be fine ...

**Pull request merged**